assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js

```diff
@@ -297,71 +286,97 @@ export default function UserDimensionsPieChart( { dimensionName, dimensionValue,
+									'googlesitekit-widget--analyticsAllTraffic__legend-slice',
+									{
+										'googlesitekit-widget--analyticsAllTraffic__legend-active': isActive,
+										'googlesitekit-widget--analyticsAllTraffic__legend-others': isOthers,
+									}
+								) }
+							>
+								<span className="googlesitekit-widget--analyticsAllTraffic__dot" style={ { backgroundColor: sliceColor } } />
+
+								<span className="googlesitekit-widget--analyticsAllTraffic__label" data-label={ label }>
+									{ label }
+								</span>
+
+								<span className="googlesitekit-widget--analyticsAllTraffic__underlay" style={ { backgroundColor: sliceColor } } />
+							</Link>
+						);
+					} ) }
+
+					{ ! loaded && (
+						<Fragment>
+							<span className="googlesitekit-widget--analyticsAllTraffic__legend-slice">
+								<span className="googlesitekit-widget--analyticsAllTraffic__dot" style={ { backgroundColor: '#ccc' } } />
+								<PreviewBlock height="18px" width="68px" shape="square" />
+							</span>
 
-			<div className="googlesitekit-widget--analyticsAllTraffic__legend">
-				{ dataMap?.slice( 1 ).map( ( [ label ], i ) => {
-					const isActive = label === dimensionValue;
-					const sliceColor = slices[ i ]?.color;
-					const isOthers = __( 'Others', 'google-site-kit' ) === label;
-
-					return (
-						<Link
-							key={ label }
-							onClick={ () => onLegendClick( i ) }
-							className={ classnames(
-								'googlesitekit-widget--analyticsAllTraffic__legend-slice',
-								{
-									'googlesitekit-widget--analyticsAllTraffic__legend-active': isActive,
-									'googlesitekit-widget--analyticsAllTraffic__legend-others': isOthers,
-								}
-							) }
-						>
-							<span className="googlesitekit-widget--analyticsAllTraffic__dot" style={ { backgroundColor: sliceColor } } />
-
-							<span className="googlesitekit-widget--analyticsAllTraffic__label" data-label={ label }>
-								{ label }
+							<span className="googlesitekit-widget--analyticsAllTraffic__legend-slice">
+								<span className="googlesitekit-widget--analyticsAllTraffic__dot" style={ { backgroundColor: '#ccc' } } />
+								<PreviewBlock height="18px" width="52px" shape="square" />
 							</span>
 
-							<span className="googlesitekit-widget--analyticsAllTraffic__underlay" style={ { backgroundColor: sliceColor } } />
-						</Link>
-					);
-				} ) }
+							<span className="googlesitekit-widget--analyticsAllTraffic__legend-slice">
+								<span className="googlesitekit-widget--analyticsAllTraffic__dot" style={ { backgroundColor: '#ccc' } } />
+								<PreviewBlock height="18px" width="40px" shape="square" />
+							</span>
+
+							<span className="googlesitekit-widget--analyticsAllTraffic__legend-slice">
+								<span className="googlesitekit-widget--analyticsAllTraffic__dot" style={ { backgroundColor: '#ccc' } } />
+								<PreviewBlock height="18px" width="52px" shape="square" />
+							</span>
+						</Fragment>
```

Comment:
How about this?

```javascript
new Array( 4 ).fill().map( ( _, i ) => (
    <span key={ i } className="googlesitekit-widget--analyticsAllTraffic__legend-slice">
        <span className="googlesitekit-widget--analyticsAllTraffic__dot" style={ { backgroundColor: '#ccc' } } />
        <PreviewBlock height="18px" width="68px" shape="square" />
    </span>
) )
```

Reply:
I was going to, but the varied sizes create a slightly more "natural" loading size (usually the four dimension names have different widths, so making them all the same width during loading feels less natural.)

A bonus is that for some locals these will line up perfectly with the actual loaded data, but even when they don't: the variance looks a bit nicer 🙂

Reply:
@tofumatt ok, then how about this?

```javascript
[ 68, 52, 40, 52  ].map( ( width, i ) => (
    <span key={ i } className="googlesitekit-widget--analyticsAllTraffic__legend-slice">
        <span className="googlesitekit-widget--analyticsAllTraffic__dot" style={ { backgroundColor: '#ccc' } } />
        <PreviewBlock height="18px" width={ `${ width }px` } shape="square" />
    </span>
) )
```